src/MvvmNano.Forms/BetterMasterDetailPage.cs

```diff
@@ -0,0 +1,112 @@
+    /// </summary>
+    public class BetterMasterDetailPage :  MasterDetailPage
+    {
+        private MvvmNanoApplication Application => Xamarin.Forms.Application.Current as MvvmNanoApplication;
+
+        private MvvmNanoFormsPresenter Presenter => (MvvmNanoFormsPresenter)MvvmNanoIoC.Resolve<IPresenter>();
+ 
+        /// <summary>
+        /// This ListView contains all detail entrys. Add it to your custom master.
+        /// </summary>
+        protected ListView DetailListView { get; private set; }= new ListView();
+
+        /// <summary>
+        /// ContentPage that represents the menu.
+        /// </summary>
+        protected new ContentPage Master => base.Master as ContentPage;
+
+        /// <summary>
+        /// MasterContent of the menu.
+        /// </summary>
+        public View MasterContent
+        {
+            get { return Master.Content; }
+            set { Master.Content = value; }
+        }
+
+        public BetterMasterDetailPage()
+        {
+            DetailListView.ItemsSource = Application.MasterDetails;
+            DetailListView.ItemTemplate = new DataTemplate(() =>
+            {
+                Label titleLabel = new Label
+                {
+                    VerticalOptions = LayoutOptions.Center,
+                    HorizontalOptions = LayoutOptions.Center
+                };
+                titleLabel.SetBinding(Label.TextProperty, "Title");
+                return new ViewCell
+                {
+                    View = titleLabel
+                };
+            });
+            base.Master = new ContentPage() {Title = "Master", Content = DetailListView}; 
+            Detail = new ContentPage() {Title = "Default content page."};
+            DetailListView.ItemSelected += MenuEntrySelected;
+            DetailListView.ItemTapped += MenuEntryTapped;
+        }  
+        
+
+        /// <summary>
+        /// Hide the menu if an item is tapped that is already selected. 
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="itemTappedEventArgs"></param>
+        private void MenuEntryTapped(object sender, ItemTappedEventArgs itemTappedEventArgs)
+        {
+            if (itemTappedEventArgs.Item == DetailListView.SelectedItem)
+                IsPresented = false;
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="selectedItemChangedEventArgs"></param>
+        private void MenuEntrySelected(object sender, SelectedItemChangedEventArgs selectedItemChangedEventArgs)
+        {
+            var data = DetailListView.SelectedItem as MasterDetailData;
+            Presenter.SetDetail(data);
+        }
+
+        /// <summary>
+        /// All events: ABOARD THE SINKING PAGE!!!
+        /// </summary>
+        protected override void OnDisappearing()
+        {
+            DetailListView.ItemSelected -= MenuEntrySelected;
```

Comment:
Are you sure this is the right place do detach the handlers? You're just attaching them once in the constructor, but I would guess this method could be called multiple times (for example whenever something is being pushed to the navigation stack on top of that page).